### PR TITLE
[TwigBundle] remove legacy template dirs from iterator

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | yes
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Some leftovers from #31667

### CODE REVIEW

File: src/Symfony/Bundle/TwigBundle/TemplateIterator.php [hunk 1]
```diff
@@ -22,21 +22,18 @@
...
     private $defaultPath;
 
     /**
      * @param KernelInterface $kernel      A KernelInterface instance
-     * @param string          $rootDir     The directory where global templates can be stored
      * @param array           $paths       Additional Twig paths to warm
      * @param string          $defaultPath The directory where global templates can be stored
      */
-    public function __construct(KernelInterface $kernel, string $rootDir, array $paths = [], string $defaultPath = null)
+    public function __construct(KernelInterface $kernel, array $paths = [], string $defaultPath)
```
Comment (Tobion) [inline src/Symfony/Bundle/TwigBundle/TemplateIterator.php]: In theory it's missing a deprecation in 4.4. But there are several compiler passes that expect the paths at a certain argument number. So either we update the compiler passes like done in this PR but that might not work when people overwrote the definition. Or we don't update the compiler passes which would trigger internal deprecations.
The only solution I see is to mark this class as `@internal`. To me this class is an implementation detail and should not be relied on as an extension point. This I guess is also why `TemplateCacheWarmer` does not typhint this class but `iterable`.

Comment (Tobion) [reply src/Symfony/Bundle/TwigBundle/TemplateIterator.php]: #31828

**Comment by Tobion** [comment]
Waiting on #31828 and then rebasing

**Comment by fabpot** [comment]
@Tobion #31828 has been merged now.

**Comment by Tobion** [comment]
Rebased

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Tobion.

**Pull request merged**